exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/SpanAdapter.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.otlp;
+
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span;
+import io.opentelemetry.proto.trace.v1.Span.SpanKind;
+import io.opentelemetry.proto.trace.v1.Status;
+import io.opentelemetry.proto.trace.v1.Status.StatusCode;
+import io.opentelemetry.sdk.contrib.otproto.TraceProtoUtils;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.SpanData.TimedEvent;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Link;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+final class SpanAdapter {
+  static List<ResourceSpans> toProtoResourceSpans(List<SpanData> spanDataList) {
+    Map<Resource, ResourceSpans.Builder> resourceSpansBuilderMap = new HashMap<>();
+    for (SpanData spanData : spanDataList) {
+      Resource resource = spanData.getResource();
+      ResourceSpans.Builder resourceSpansBuilder =
+          resourceSpansBuilderMap.get(spanData.getResource());
+      if (resourceSpansBuilder == null) {
+        resourceSpansBuilder =
+            ResourceSpans.newBuilder().setResource(ResourceAdapter.toProtoResource(resource));
+        resourceSpansBuilderMap.put(resource, resourceSpansBuilder);
```

Comment:
Can we really have more than one Resource per SDK? I wonder if we should pass the Resource in at the top level of the export API, rather than have to make every exporter do this convoluted logic.

Reply:
We can have multiple instances of the SDK, but not multiple Resources per SDK. Maybe we should file an issue for this.

Reply:
ok. Let's extract this bit that builds the map into it's own method, for clarity.

Reply:
I think the logic is pretty isolated, can you clarify how would you like the code to be structured?

Reply:
I'd just take the code that creates the Map and populates it and move it to its own method.

Reply:
While I am constructing the Map I am also converting the Spans and add them to the builder. So not sure how to cleanly separate them.

Reply:
couldn't lines 37 through 48 be a method that returns the `Map<Resource, ResourceSpans.Builder>`?

Reply:
Yes but that methods also converts Spans so for me it is not that of a clear separation.

Reply:
But if that unblocks this PR I will do it :)

Reply:
I think that's fair. I won't hold it up if you want to keep it as-is. We have bigger fish to fry. :)